prime-angular-frontend/src/app/shared/components/address-autocomplete/address-autocomplete.component.ts

```diff
@@ -32,40 +36,44 @@ export class AddressAutocompleteComponent implements OnInit {
   }
 
   public onAutocomplete(id: string) {
-    this.addressRetrieved = null;
     this.addressAutocompleteResource.retrieve(id)
-      .subscribe((response: AddressAutocompleteRetrieveResponse[]) => {
-        response.forEach((field) => {
-          if (field.language === 'ENG') {
-            this.addressRetrieved = field;
-          }
-        });
-        const address = new Address();
-        address.countryCode = this.addressRetrieved.countryIso2;
-        address.provinceCode = this.addressRetrieved.provinceCode;
-        address.city = this.addressRetrieved.city;
-        address.street = this.addressRetrieved.line1;
-        address.street2 = this.addressRetrieved.line2;
-        address.postal = this.addressRetrieved.postalCode;
+      .subscribe((results: AddressAutocompleteRetrieveResponse[]) => {
+        this.addressRetrieved = results.find(result => result.language === 'ENG') ?? null;
+
+        if (this.addressRetrieved) {
+          const address = new Address(
+            this.addressRetrieved.countryIso2,
+            this.addressRetrieved.provinceCode,
+            this.addressRetrieved.line1,
+            this.addressRetrieved.line2,
+            this.addressRetrieved.city,
+            this.addressRetrieved.postalCode
+          );
+
+          (!this.bcOnly || address.provinceCode === 'BC')
+            ? this.autocompleteAddress.emit(address)
+            : this.toastService.openErrorToast('Address must be located in BC');
 
-        (!this.bcOnly || address.provinceCode === 'BC')
-          ? this.autocompleteAddress.emit(address)
-          : this.toastService.openErrorToast('Address must be in BC');
+        } else {
+          this.toastService.openErrorToast('Address could not be retrieved');
+        }
       });
   }
 
   public ngOnInit(): void {
-    this.form = new FormGroup({ autocomplete: new FormControl() });
+    this.form = this.fb.group({ autocomplete: [''] });
 
     this.autocomplete.valueChanges
-      .subscribe(() => {
-        this.addressAutocompleteFields = [];
-        if (this.autocomplete.value) {
-          this.addressAutocompleteResource.find(this.autocomplete.value)
```

Comment:
@AnaisHebert observables don't like to be in observables.  This is an anti-pattern that can result in some issues, but will most likely work 90% of the time.  You want to switch what you're observing using operators like `exhaustMap`, and then block if not further processing is required using `EMPTY`